pyiron_base/interfaces/has_dict.py

```diff
@@ -66,7 +109,17 @@ def _type_to_dict(self):
         return type_dict
 
     def to_dict(self):
-        return self._to_dict() | self._type_to_dict()
+        type_dict = self._type_to_dict()
+        data_dict = {}
+        child_dict = {}
+        for k, v in self._to_dict().items():
+            if isinstance(v, HasDict):
+                child_dict[k] = v.to_dict()
+            elif isinstance(v, HasHDF):
+                child_dict[k] = HasDictfromHDF.to_dict(v)
+            else:
+                data_dict[k] = v
```

Comment:
Just to clarify this, this part currently does not work for the `Atoms` object in `pyiron_atomistics`, as it is neither derived from `HasDict` nor from `HasHDF`, correct? It only works when the `Atoms` object was stored as part of a `DataContainer`, correct?